src/components/components/ebay-tooltip-base/index.js

```diff
@@ -0,0 +1,173 @@
+const Expander = require('makeup-expander');
+const emitAndFire = require('../../../common/emit-and-fire');
+const template = require('./template.marko');
+
+function getInitialState(input) {
+    input.location = input.location || 'bottom';
+
+    return input;
+}
+
+function getTemplateData(state) {
+    return state;
+}
+
+function init() {
+    const isHostPresent = this.el.querySelector(`.${this.state.type}__host`);
+
+    if (isHostPresent) {
+        this.expander = new Expander(this.el, {
+            hostSelector: `.${this.state.type}__host`,
+            contentSelector: `.${this.state.type}__overlay`,
+            focusManagement: this.state.type === 'tourtip' ? null : 'focusable',
+            expandOnFocus: this.state.type === 'tooltip',
+            expandOnHover: this.state.type === 'tooltip',
+            expandOnClick: this.state.type === 'infotip',
+            autoCollapse: this.state.type === 'tooltip'
+        });
+    }
+
+    this.alignOverlay();
+}
+
+function handleExpand() {
+    this.alignOverlay();
+    this.setState('expanded', true);
+    emitAndFire(this, 'tooltip-expand');
+}
+
+function handleCollapse() {
+    this.setState('expanded', false);
+    emitAndFire(this, 'tooltip-collapse');
+}
+
+function handleTooltipClose() {
+    this.setState('expanded', false);
+    this.expander.collapse();
+    emitAndFire(this, 'tooltip-close');
+}
+
+function alignOverlay() {
+    const overlay = this.el.querySelector(`.${this.state.type}__overlay`);
```

Comment:
Since all of these styles are now static it would be best we just did this mapping in `getTemplateData`. It will simplify the logic (no need to re apply this on init or expand) and will mean that it will work with server side rendering for tooltips that should open by default.

Reply:
`this.el` is `undefined` should I be using `document.querySelector` (or something else)?

Reply:
Yeah, I don't understand how this will work in getTemplateData, because I need a DOM element which has already been added to the page. getTemplateData doesn't seem to have any Marko DOM element references for me to tack onto. Am I missing something?

Reply:
`this` is not available in `getTemplateData`. In this case you should only need `state` which you can access like:

```
getTemplateData(state, input) { ... }
```

Reply:
I'm saying we can create the style data in `getTemplateData` and forward that to the template and assign the `style` attribute there, rather than directly mutating the overlay.

Reply:
As a string/object/array. Okay, I think I get it. I was moving that direction.